test/unit/services/listeners/edition_dependencies_test.rb

```diff
@@ -0,0 +1,45 @@
+require "test_helper"
+
+class ServiceListeners::EditionDependenciesTest < ActiveSupport::TestCase
+
+  [:publisher, :force_publisher].each do |service_name|
+    test "Whitehall.edition_services.#{service_name} populates edition's dependencies" do
+      contact, speech = create(:contact), create(:speech)
+      news_article = create(:submitted_news_article, body: "For more information, get in touch at:
+        [Contact:#{contact.id}] or read our [official statement](/government/admin/speeches/#{speech.id})", major_change_published_at: Time.zone.now)
+
+      stub_panopticon_registration(news_article)
+
+      assert Whitehall.edition_services.send(service_name, news_article).perform!
+      assert_equal [contact], news_article.depended_upon_contacts
+      assert_equal [speech], news_article.depended_upon_editions
+    end
+
+    test "Whitehall.edition_services.#{service_name} cleans-up a dependable edition's dependency records" do
```

Comment:
I think this could be made a bit clearer. In the previous test you are (force)publishing an edition and expecting that its list of dependencies are updated. In this test you are (force)publishing a depended-upon edition and expecting that it is no longer depended upon. I think it would be helpful to make it very clear that the thing which is being published is at a different end of the dependency relationship. 

For example:

```
["", "force"].each do |force| 
  test "#{force} publishing a depended-upon edition removes it as a dependency" do
    service_name = force == "" ? :publisher : :force_publisher
    ...
  end
end
```